### DISCUSSION TITLE
Vnet Associations and Reservations

### DISCUSSION THREAD

**Discussion by positivism1** [Azure/ipam#162] [General]
Hello 

I am requesting a vnet cidr using NextAvailableVnet, this creations a reservation. 
I use this request to create a vnet, when I try  and associate this vnet to the block I get this error:
![image](https://github.com/Azure/ipam/assets/3076159/954da9f3-921a-42d6-9529-8c3fec6b72dc)

Am I just using this app wrong? I want the vnet associated to the block so it looks good in the UI 🤣

**Comment by DCMattyG** [comment]
Good morning @positivism1, the very high level process is this:

1. Create a Reservation from the Block, this will give you a CIDR and a Tag for the new Virtual Network
2. Create the new Virtual Network in Azure, using the CIDR from the Reservation and applying the Tag
3. Wait approximately 1-2 minutes, the IPAM tool will automatically find the new Virtual Network and associate to the correct Block which it knows because of the Tag ID

In the above example, did you apply the Tag to the new Virtual Network? If so, the tool may have already completed the association, hence the message you are seeing.

**Reply by positivism1** [reply]
Ahh I have to wait? I'm really good at waiting. 

Thank you! apologies for all the questions. I will give "Waiting" a go. 👍 🤣
